test/integration/standalone/other/tls_test.go

```diff
@@ -74,4 +81,203 @@ var _ = g.Describe("Query service_cpm_minute", func() {
+
+		// Read original certificate and key
+		originalCert, err := os.ReadFile(certFile)
+		gm.Expect(err).NotTo(gm.HaveOccurred())
+		originalKey, err := os.ReadFile(keyFile)
+		gm.Expect(err).NotTo(gm.HaveOccurred())
+
+		// Write to temporary location
+		err = os.WriteFile(tempCertFile, originalCert, 0o600)
+		gm.Expect(err).NotTo(gm.HaveOccurred())
+		err = os.WriteFile(tempKeyFile, originalKey, 0o600)
+		gm.Expect(err).NotTo(gm.HaveOccurred())
+
+		// Start a new server using the temporary certificate files
+		tempAddr, _, tempDeferFn := setup.StandaloneWithTLS(tempCertFile, tempKeyFile)
+		defer tempDeferFn()
+
+		// Create initial connection with the original certificate
+		creds, err := credentials.NewClientTLSFromFile(tempCertFile, "localhost")
+		gm.Expect(err).NotTo(gm.HaveOccurred())
+		tempConn, err := grpchelper.Conn(tempAddr, 10*time.Second, grpclib.WithTransportCredentials(creds))
+		gm.Expect(err).NotTo(gm.HaveOccurred())
+		defer tempConn.Close()
+
+		// Populate test data and verify with original certificate connection first
+		ns := timestamp.NowMilli().UnixNano()
+		testBaseTime := time.Unix(0, ns-ns%int64(time.Minute))
+		casesMeasureData.Write(tempConn, "service_cpm_minute", "sw_metric", "service_cpm_minute_data.json", testBaseTime, interval)
+
+		// Verify using the initial connection before updating certificates
+		gm.Eventually(func(innerGm gm.Gomega) {
+			casesMeasureData.VerifyFn(innerGm, helpers.SharedContext{
+				Connection: tempConn,
+				BaseTime:   testBaseTime,
+			}, helpers.Args{Input: "all", Duration: 25 * time.Minute, Offset: -20 * time.Minute})
+		}, flags.EventuallyTimeout).Should(gm.Succeed())
+
+		// Generate a new certificate with a different CommonName
+		certPEM, keyPEM, err := tls.GenerateSelfSignedCert("updated-localhost", []string{"localhost"})
+		gm.Expect(err).NotTo(gm.HaveOccurred())
+
+		// Update the certificate files in the temporary location
+		err = os.WriteFile(tempCertFile, certPEM, 0o600)
+		gm.Expect(err).NotTo(gm.HaveOccurred())
+		err = os.WriteFile(tempKeyFile, keyPEM, 0o600)
+		gm.Expect(err).NotTo(gm.HaveOccurred())
+
+		// Wait for the server to reload the certificates
+		time.Sleep(1 * time.Second)
+
+		// Create a new connection with the updated certificates
+		newCreds, err := credentials.NewClientTLSFromFile(tempCertFile, "updated-localhost")
+		gm.Expect(err).NotTo(gm.HaveOccurred())
+		newConn, err := grpchelper.Conn(tempAddr, 10*time.Second, grpclib.WithTransportCredentials(newCreds))
+		gm.Expect(err).NotTo(gm.HaveOccurred())
+		defer newConn.Close()
+
+		// Verify using the connection with new certificates
+		gm.Eventually(func(innerGm gm.Gomega) {
+			casesMeasureData.VerifyFn(innerGm, helpers.SharedContext{
+				Connection: newConn,
+				BaseTime:   testBaseTime,
+			}, helpers.Args{Input: "all", Duration: 25 * time.Minute, Offset: -20 * time.Minute})
+		}, flags.EventuallyTimeout).Should(gm.Succeed())
+	})
+
+	g.It("queries an updated HTTP server", func() {
```

Comment:
You can replace this test with the previous one since it verifies both HTTP and gRPC.